crates/mcp-client/src/transport/stdio.rs

```diff
@@ -0,0 +1,190 @@
+use std::sync::Arc;
+use tokio::process::{Child, ChildStdin, ChildStdout, Command};
+
+use async_trait::async_trait;
+use mcp_core::protocol::JsonRpcMessage;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::sync::{mpsc, oneshot, Mutex};
+use tokio::task::JoinHandle;
+
+use super::{Error, Transport, TransportMessage};
+
+/// A `StdioTransport` uses a child process's stdin/stdout as a communication channel.
+///
+/// It uses channels for message passing and handles responses asynchronously through a background task.
+pub struct StdioTransport {
+    command: String,
+    args: Vec<String>,
+    process: Arc<Mutex<Option<Child>>>,
+    reader_handle: Arc<Mutex<Option<JoinHandle<()>>>>,
+    pending_requests: Arc<
+        Mutex<std::collections::HashMap<String, oneshot::Sender<Result<JsonRpcMessage, Error>>>>,
+    >,
+}
+
+impl StdioTransport {
+    /// Create a new `StdioTransport` configured to run the given command with arguments.
+    pub fn new<S: Into<String>>(command: S, args: Vec<String>) -> Self {
+        Self {
+            command: command.into(),
+            args,
+            process: Arc::new(Mutex::new(None)),
+            reader_handle: Arc::new(Mutex::new(None)),
+            pending_requests: Arc::new(Mutex::new(std::collections::HashMap::new())),
+        }
+    }
+
+    async fn spawn_process(&self) -> Result<(ChildStdin, ChildStdout), Error> {
+        let mut child = Command::new(&self.command)
```

Comment:
should add https://tikv.github.io/doc/tokio/process/struct.Command.html#method.kill_on_drop here i think?